### PR TITLE
Integrated RainbowKit

### PR DESCRIPTION
**Pull request by benbaessler**
## Description

Closes [#151](https://github.com/lenstube-xyz/lenstube/issues/151)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

### CODE REVIEW

File: src/components/Common/Providers.tsx [hunk 1]
```diff
@@ -25,29 +27,41 @@ const { chains, provider } = configureChains(
...
-        jsonRpcUrl: POLYGON_RPC_URL
-      }
-    })
-  ]
-}
+const connectors = connectorsForWallets([
+  {
+    groupName: 'Recommended',
+    wallets: [
+      metaMaskWallet({ chains, shimDisconnect: true }),
```
Comment (sasicodes) [inline src/components/Common/Providers.tsx:34]: injectedWallet can be here (eg. brave / any other connectors)

Comment (sasicodes) [reply src/components/Common/Providers.tsx:34]: https://www.rainbowkit.com/docs/custom-wallet-list

File: src/components/Common/Providers.tsx [hunk 2]
```diff
@@ -58,11 +72,13 @@ const wagmiClient = createClient({
 const Providers = ({ children }: { children: ReactNode }) => {
   return (
     <WagmiConfig client={wagmiClient}>
-      <ApolloProvider client={apolloClient}>
-        <ThemeProvider defaultTheme="light" attribute="class">
-          {children}
-        </ThemeProvider>
-      </ApolloProvider>
+      <RainbowKitProvider chains={chains}>
```
Comment (sasicodes) [inline src/components/Common/Providers.tsx:75]: dark mode support if possible

Comment (benbaessler) [reply src/components/Common/Providers.tsx]: Added

File: src/components/Common/Auth/ConnectWalletButton.tsx [hunk 3]
```diff
@@ -47,6 +48,8 @@ const ConnectWalletButton = ({ handleSign, signing }: Props) => {
 
   const { mounted } = useIsMounted()
 
+  const { openConnectModal } = useConnectModal()
+
```
Comment (sasicodes) [inline src/components/Common/Auth/ConnectWalletButton.tsx:52]: we can safely get rid of <Modal> in this component and related unused states and functions 🙇🏻

Comment (benbaessler) [reply src/components/Common/Auth/ConnectWalletButton.tsx:30]: Should be all cleaned up now 🙂

**Pull request merged**